src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -908,53 +799,34 @@ namespace OpenLoco::Ui::Windows::PromptBrowse
                 return;
             }
 
-            // Create full path to target file.
-            fs::path path = fs::u8path(&_directory[0]) / inputSession.buffer;
-
-            // Append extension to filename.
-            path += getExtensionFromFileType(_fileType);
-
             // Does the file already exist?
             if (fs::exists(path))
             {
                 // Copy directory and filename to buffer.
                 char* buffer_2039 = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
                 strncpy(&buffer_2039[0], inputSession.buffer.c_str(), 512);
 
+                // Arguments for description text in ok/cancel window.
                 FormatArguments args{};
                 args.push(StringIds::buffer_2039);
 
-                // Copy window title into title buffer for ok/cancel window.
-                strncpy(&_stringFormatBuffer[0], _title, 256);
-
-                // Formatted string into description buffer for ok/cancel window.
-                loco_global<char[512], 0x0112CE04> descriptionBuffer;
-                StringManager::formatString(&descriptionBuffer[0], StringIds::replace_existing_file_prompt, &args);
-
                 // Ask for confirmation to replace the file.
-                if (!Windows::PromptOkCancel::open(StringIds::replace_existing_file_button))
+                auto titleId = self->widgets[widx::caption].text;
+                if (!Windows::PromptOkCancel::open(titleId, StringIds::replace_existing_file_prompt, args, StringIds::replace_existing_file_button))
                     return;
             }
-
-            // Copy directory and filename to buffer.
-            strncpy(&_directory[0], path.u8string().c_str(), std::size(_directory));
-        }
-        else
-        {
-            // Copy scenario path into expected address. Trailing / on directory is assumed.
-            // TODO: refactor to fs::path?
-            strncat(_directory, inputSession.buffer.c_str(), std::size(_directory));
-            strncat(_directory, getExtensionFromFileType(_fileType), std::size(_directory));
         }
 
+        // Copy directory and filename to buffer.
+        strncpy(_savePath.get(), path.u8string().c_str(), std::size(_savePath));
+
         // Remember the current path for saved games
-        if (_fileType == browse_file_type::saved_game)
+        if (_fileType == BrowseFileType::savedGame)
         {
-            fs::path currentPath = fs::u8path(&_directory[0]);
-            if (!fs::is_directory(currentPath))
-                Config::getNew().last_save_path = currentPath.parent_path().u8string();
+            if (!fs::is_directory(_currentDirectory))
```

Comment:
um what? if directory is not directory?

Reply:
You're right, this check is no longer needed. In vanilla, it can contain a file path a well.